sql/core/src/test/scala/org/apache/spark/sql/scripting/SqlScriptingIntegrationSuite.scala

```diff
@@ -0,0 +1,153 @@
+
+class SqlScriptingIntegrationSuite extends QueryTest with SharedSparkSession {
+  // Helpers
+  private def verifySqlScriptResult(
+      sqlText: String, expected: Seq[Seq[Row]], printResult: Boolean = false): Unit = {
+    val interpreter = SqlScriptingInterpreter()
+    val compoundBody = spark.sessionState.sqlParser.parseScript(sqlText)
+    val executionPlan = interpreter.buildExecutionPlan(compoundBody)
+    val result = executionPlan.flatMap {
+      case statement: SingleStatementExec =>
+        if (printResult) {
+          // scalastyle:off println
+          println("Executing: " + statement.getText(sqlText))
+          // scalastyle:on println
+        }
+
+        if (statement.consumed) {
+          None
+        } else {
+          Some(Dataset.ofRows(spark, statement.parsedPlan, new QueryPlanningTracker))
+        }
+      case _ => None
+    }.toArray
+
+    assert(result.length == expected.length)
+    result.zip(expected).foreach{ case (df, expectedAnswer) => checkAnswer(df, expectedAnswer)}
+  }
+
+  // Tests
+  test("select 1") {
+    verifySqlScriptResult("SELECT 1;", Seq(Seq(Row(1))))
+  }
+
+  test("multi statement - simple") {
+    withTable("t") {
+      val sqlScript =
+        """
+          |BEGIN
+          |CREATE TABLE t (a INT, b STRING, c DOUBLE) USING parquet;
+          |INSERT INTO t VALUES (1, 'a', 1.0);
+          |SELECT a, b FROM t WHERE a = 12;
+          |SELECT a FROM t;
+          |END
+          |""".stripMargin
+      val expected = Seq(
+        Seq.empty[Row], // create table
+        Seq.empty[Row], // insert
+        Seq.empty[Row], // select with filter
+        Seq(Row(1))
+      )
+      verifySqlScriptResult(sqlScript, expected)
+    }
+  }
+
+  test("multi statement - count") {
+    withTable("t") {
+      val sqlScript =
+        """
+          |BEGIN
+          |CREATE TABLE t (a INT, b STRING, c DOUBLE) USING parquet;
+          |INSERT INTO t VALUES (1, 'a', 1.0);
+          |INSERT INTO t VALUES (1, 'a', 1.0);
+          |SELECT
+          | CASE WHEN COUNT(*) > 10 THEN true
+          | ELSE false
+          | END AS MoreThanTen
+          |FROM t;
+          |END
+          |""".stripMargin
+      val expected = Seq(
+        Seq.empty[Row], // create table
+        Seq.empty[Row], // insert #1
+        Seq.empty[Row], // insert #2
+        Seq(Row(false))
+      )
+      verifySqlScriptResult(sqlScript, expected)
+    }
+  }
+
+  test("session vars - set and read") {
+    val sqlScript =
+      """
+        |BEGIN
+        |DECLARE var = 1;
+        |SET VAR var = var + 1;
+        |SELECT var;
+        |END
+        |""".stripMargin
+    val expected = Seq(
+      Seq.empty[Row], // declare var
+      Seq.empty[Row], // set var
+      Seq(Row(2)), // select
+      Seq.empty[Row] // drop var
+    )
+    verifySqlScriptResult(sqlScript, expected)
+  }
+
+  test("session vars - set and read scoped") {
+    val sqlScript =
+      """
+        |BEGIN
```

Comment:
TODO (in next PR) is a test that:
1) creates a var
2) does some operation that throws an error (e.g. insert against constraint/create table that already exists...)
3) assert that variable is no longer in session after error is thrown.

This currently is not going to work since we manually insert DROPs.

Reply:
Agreed, we already have an item that includes this topic: https://issues.apache.org/jira/browse/SPARK-48530.